src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -492,6 +492,10 @@ void MacroAssembler::print_state() {
 
 // 64 bit versions
 
+void MacroAssembler::reset_last_Java_frame(bool clear_fp) {
+  reset_last_Java_frame(r15_thread, clear_fp);
+}
```

Comment:
I see that before the Foreign Linker integration, this method was between `MacroAssembler::pushptr` and `MacroAssembler::set_last_Java_frame`. Please move it back there.